data-pipeline/src/trace_exporter.rs

```diff
@@ -253,15 +254,14 @@ impl TraceExporter {
 
     /// Send msgpack serialized traces to the agent
     #[allow(missing_docs)]
-    pub fn send(&self, data: &[u8], trace_count: usize) -> Result<String, String> {
+    pub fn send<T>(&self, data: T, trace_count: usize) -> Result<String, String>
+    where
+        T: UnderlyingBytes,
```

Comment:
I think that this code should just accept `tinybytes::Bytes` and it's up to the caller to figure out how to provide that. It would also mean that we don't have to implement `UnderlyingBytes` for the `ByteSlice`

Reply:
> It would also mean that we don't have to implement UnderlyingBytes for the ByteSlice

Because we would just call `tinybytes::Bytes::from_static` in `ddog_trace_exporter_send` like @VianneyRuhlmann [was asking](https://github.com/DataDog/libdatadog/pull/764#discussion_r1860570571)? 

I guess at the end of the day `send` being generic doesn't really save the caller anything. I can make that change.

Reply:
done